modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -222,16 +222,25 @@ struct IEUnit {
     IE::ExecutableNetwork this_network;
     cv::gimpl::ie::wrap::Plugin this_plugin;
 
+    InferenceEngine::RemoteContext::Ptr rctx = nullptr;
```

Comment:
So now a question. Should different `infer<>()` calls in the same graph share the same context object or have their own? What happens if they have their own, but created with the same `context params` ? Would it result into the same context being used or we end up with N different contexts for the same device?

Reply:
It seems like it's supposed to be on the same level as `IE::Core` but also 1 per thread/process

Reply:
Answering your question, I think different `infer`s should share the same context within the same graph

Reply:
On the second thought, I don't think it really matters how to store `RemoteContext` here - `context params` is the thing that defines the behavior

Reply:
so do they share it or not? Remember every `infer<>` call in the graph gets its own `IEUnit` structure.

Reply:
Actually, for every infer we can configure it differently in terms of `context params`, so they might not share it.

Reply:
exactly, but is it OK that we create multiple `RemoteContext` objects with the same `params`, that's the question.

Anyway, at these level we can't say for sure if `params` are really same or not, so let's fix it later if the issue is relevant.